extensions/ql-vscode/src/helpers.ts

```diff
@@ -785,3 +785,35 @@ export async function* walkDirectory(
     }
   }
 }
+
+/**
+ * Returns the path of the first folder in the workspace.
+ * This is used to decide where to create skeleton QL packs.
+ *
+ * If the first folder is a QL pack, then the parent folder is returned.
+ * This is because the vscode-codeql-starter repo contains a ql pack in
+ * the first folder.
+ *
+ * This is a temporary workaround until we can retire the
+ * vscode-codeql-starter repo.
+ */
+
+export function getFirstWorkspaceFolder() {
+  const workspaceFolders = getOnDiskWorkspaceFolders();
+
+  if (!workspaceFolders || workspaceFolders.length === 0) {
+    throw new Error("No workspace folders found");
+  }
+
+  const firstFolderFsPath = workspaceFolders[0];
+
+  // For the vscode-codeql-starter repo, the first folder will be a ql pack
+  // so we need to get the parent folder
+  if (firstFolderFsPath.includes("codeql-custom-queries")) {
+    // return the parent folder
+    return dirname(firstFolderFsPath);
```

Comment:
Is the issue really that the first folder is a qlpack, not that we are part of the `vscode-codeql-starter` repo? If that's the case, would it be better to check if there is a `qlpack.yml` or `codeql-pack.yml` file in the specified directory?